### PR TITLE
Transfer back the debt to the Proxy owner

### PR DESCRIPTION
**Pull request by theonlyfoxy**

### CODE REVIEW

File: contracts/Proxy/BorrowerOperationsScript.sol [hunk 1]
```diff
@@ -4,42 +4,88 @@ pragma solidity 0.6.11;
 
 import "../Dependencies/CheckContract.sol";
 import "../Interfaces/IBorrowerOperations.sol";
+import "../Dependencies/IERC20.sol";
 
 
-contract BorrowerOperationsScript is CheckContract {
```
Comment (bingen) [inline contracts/Proxy/BorrowerOperationsScript.sol:9]: Why are you removing this? Shouldn’t you remove the import above too?

File: contracts/Proxy/BorrowerOperationsScript.sol [hunk 2]
```diff
@@ -4,42 +4,88 @@ pragma solidity 0.6.11;
...
+    address immutable LUSD;
 
-    constructor(IBorrowerOperations _borrowerOperations) public {
-        checkContract(address(_borrowerOperations));
+    constructor(IBorrowerOperations _borrowerOperations, address _LUSD) public {
         borrowerOperations = _borrowerOperations;
+        LUSD = _LUSD;
     }
 
-    function openTrove(uint _maxFee, uint _LUSDAmount, address _upperHint, address _lowerHint) external payable {
```
Comment (bingen) [inline contracts/Proxy/BorrowerOperationsScript.sol:17]: Why all these format changes?

**Comment by bingen** [comment]
I’m not sure all users would want to do that. I can foresee some users wanting to keep LUSD in the proxy for other DeFi interactions through it, so I will close this for now, we can re-open it if we realize that’s wrong. Thanks anyway for your contribution.

File: contracts/Proxy/BorrowerOperationsScript.sol (hunk 2 above)
Comment (theonlyfoxy) [reply contracts/Proxy/BorrowerOperationsScript.sol:17]: Prettier...

File: contracts/Proxy/BorrowerOperationsScript.sol (hunk 1 above)
Comment (theonlyfoxy) [reply contracts/Proxy/BorrowerOperationsScript.sol:9]: Actually I forgot to remove the import. None of the inherited functions are used so any specific reason for CheckContract?

**Comment by theonlyfoxy** [comment]
> I’m not sure all users would want to do that. I can foresee some users wanting to keep LUSD in the proxy for other DeFi interactions through it, so I will close this for now, we can re-open it if we realize that’s wrong. Thanks anyway for your contribution.

Well the Maker Proxy actually does that so I am not sure if you guys would like to follow the standard.

File: contracts/Proxy/BorrowerOperationsScript.sol (hunk 1 above)
Comment (bingen) [reply contracts/Proxy/BorrowerOperationsScript.sol:9]: Well, before it was used:
```
checkContract(address(_borrowerOperations));
```
We do that as a sanity check in all of our contracts, and it was also recommended by auditors, why should we remove it?

**Comment by bingen** [comment]
That’s interesting. In which script are they doing that?

File: contracts/Proxy/BorrowerOperationsScript.sol (hunk 2 above)
Comment (bingen) [reply contracts/Proxy/BorrowerOperationsScript.sol:17]: That’s quite subjective. Some of the changed lines are as short as 36 chars, so I don’t see the need for it. Besides, it wouldn’t be consistent with the rest of our contracts.

File: contracts/Proxy/BorrowerOperationsScript.sol (hunk 1 above)
Comment (theonlyfoxy) [reply contracts/Proxy/BorrowerOperationsScript.sol:9]: I guess that was a mistake. will fix that.

File: contracts/Proxy/BorrowerOperationsScript.sol (hunk 2 above)
Comment (theonlyfoxy) [reply contracts/Proxy/BorrowerOperationsScript.sol:17]: Well that's right it doesn't match the style of the other contracts. will fix that.

**Comment by theonlyfoxy** [comment]
> That’s interesting. In which script are they doing that?

They use a separate function for that. so Liquity could do the same.
https://github.com/makerdao/dss-proxy-actions/blob/24811ae33061f0998d8306b2eee421aba6a4b39f/src/DssProxyActions.sol#L627

**Comment by theonlyfoxy** [comment]
> > That’s interesting. In which script are they doing that?
> 
> They use a separate function for that. so Liquity could do the same.
> https://github.com/makerdao/dss-proxy-actions/blob/24811ae33061f0998d8306b2eee421aba6a4b39f/src/DssProxyActions.sol#L627

Could you review this?
https://github.com/theonlyfoxy/beta/blob/patch-1/contracts/Proxy/BorrowerOperationsScript.sol

**Comment by bingen** [comment]
That one is way better! Really nice to have both option so we can cover all use cases, thanks!
One last thing, before submitting the PR for that one, I would suggest you wait until tomorrow, because as part of the launching process we are going to finish open sourcing all our code and I think there will be a repos overhaul. Sorry for the inconvenience.

**Pull request closed without merging**